### PR TITLE
Fix view distance API

### LINKED ISSUE #1349: setViewDistance() not working as expected
### What behaviour is expected:
setViewDistance() should set the view distance for a player

### What behaviour is observed:
view distance setting in spigot.yml will set the lower bound for the setting 

### Steps/models to reproduce:
in server.properties set the view distance to 2
in spigot.yml set the view distance to 2
the api can set view distance from 2

in spigot.yml set the view distance to 10
when player join, it can set to 2, soon then start load 10 chunk, after that the api cant set lower than 10

### Plugin list:
https://www.spigotmc.org/resources/set-my-view-distance.59822/

### Paper build number:
1104

### Anything else:
View distance setting in spigot.yml looks take control of mob render distance as well. 
So that whatever view distance i set using the api, the mob or player render distance will follow the setting in spigot.yml
i.e. even I can see the map for 64 chunk, but mobs i only can see 2 chunk, but i can hear the sound (so weird)
Seems like that will override the server view distance setting.

Expect the api can set spigot view distance as well, or create another lile setMobViewDistance()

### PR DESCRIPTION
**Pull request by Rickyboy320**
Aims to fix #1349 by properly using the view distance API in the EntityTracker.
Also removed the Math.max line which caused the (original) issue of not being able to set the distance lower than spigot's view distance, and replaced more occurrences of the spigot viewdistance with the API. (I believe I've got them all, directly or indirectly)

### CODE REVIEW

File: Spigot-Server-Patches/0032-Add-player-view-distance-API.patch [hunk 1]
```diff
@@ -198,12 +201,28 @@ index d975c2ccf1..6ece565c51 100644
 +            // Order matters
 +            this.setViewDistance(player, toSet);
 +            player.setViewDistance(playerViewDistance);
++
++            //Force update entity trackers
++            this.getWorld().getTracker().a(player);
```
Comment (zachbr) [inline Spigot-Server-Patches/0032-Add-player-view-distance-API.patch]: This needs an obfuscation helper.

**Review by zachbr — CHANGES_REQUESTED** [review]
Just needs an obfuscation helper. If you have any trouble with that let us know and we can handle it later.

**Comment by Rickyboy320** [comment]
Great! Added the obfuscation helper.

**Review by zachbr — APPROVED** [review]

**Comment by zachbr** [comment]
Thanks

**Pull request merged**